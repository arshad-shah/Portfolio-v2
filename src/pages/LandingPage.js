import React from "react";
import useStyles from "./styles/LandingPage.styles";
import { Box, Typography, Button, IconButton } from "@material-ui/core";
import { Link } from "react-scroll";

import ArrowDownwardRoundedIcon from "@material-ui/icons/ArrowDownwardRounded";
// icons
import GitHubIcon from "@material-ui/icons/GitHub";
import LinkedInIcon from "@material-ui/icons/LinkedIn";
import MailIcon from "@material-ui/icons/Mail";

export default function LandingPage() {
  const classes = useStyles();
  return (
    <Box className={classes.hero}>
      <Typography className={classes.typography} variant="h4">
        Hi, My Name Is
      </Typography>

      <Typography className={classes.typography} variant="h2">
        Arshad Shah!
      </Typography>

      <Typography className={classes.typography} variant="h4">
        A Second Year Computer Science Student, With a love for everything
        programming.
      </Typography>

      <Box>
        <IconButton
          role="button"
          className={classes.socialLinks}
          color="primary"
          aria-label="link to Github account of Arshad shah"
          data-block="link to Github"
          href="https://github.com/arshad-shah"
        >
          <GitHubIcon />
        </IconButton>

        <IconButton
          role="button"
          className={classes.socialLinks}
          color="primary"
          aria-label="link to Linkedin account of Arshad shah"
          data-block="link to Linkedin"
          href="https://www.linkedin.com/in/arshadshah"
        >
          <LinkedInIcon />
        </IconButton>

        <IconButton
          className={classes.socialLinks}
          color="primary"
          role="button"
          aria-label="link to mail account of Arshad shah"
          data-block="link to Mail"
          href="mailto:shaharshad1999@outlook.com"
        >
          <MailIcon />
        </IconButton>
      </Box>
      <Button
        aria-label="scroll down button"
        size="small"
        color="secondary"
        variant="outlined"
        data-block="scroll down"
      >
        <Link to="aboutme" spy={true} smooth={true}>
          <ArrowDownwardRoundedIcon />
        </Link>
      </Button>
    </Box>
  );
}
