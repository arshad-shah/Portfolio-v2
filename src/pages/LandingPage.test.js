import React from "react";
import { render, screen } from "@testing-library/react";
import LandingPage from "./LandingPage";

describe("LandingPage of the site:", () => {
  it("should render LandingPage with all the information in it", () => {
    const { container } = render(<LandingPage />);
    expect(container).toMatchSnapshot();
  });

  it("should render the Hero text", () => {
    render(<LandingPage />);
    const nameOfArshad = screen.getByText("Arshad Shah!");
    expect(nameOfArshad).toBeInTheDocument();
  });

  it("should render the scroll down button", () => {
    render(<LandingPage />);
    const scrollDownButton = screen.getByLabelText("scroll down button");
    expect(scrollDownButton).toBeInTheDocument();
  });
});
