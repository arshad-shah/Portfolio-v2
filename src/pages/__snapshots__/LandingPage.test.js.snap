// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage of the site: should render LandingPage with all the information in it 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-5 makeStyles-hero-1"
  >
    <h4
      class="MuiTypography-root makeStyles-typography-2 MuiTypography-h4"
    >
      Hi, My Name Is
    </h4>
    <h2
      class="MuiTypography-root makeStyles-typography-2 MuiTypography-h2"
    >
      Arshad Shah!
    </h2>
    <h4
      class="MuiTypography-root makeStyles-typography-2 MuiTypography-h4"
    >
      A Second Year Computer Science Student, With a love for everything programming.
    </h4>
    <div
      class="MuiBox-root MuiBox-root-6"
    >
      <a
        aria-disabled="false"
        aria-label="link to Github account of Arshad shah"
        class="MuiButtonBase-root MuiIconButton-root makeStyles-socialLinks-3 MuiIconButton-colorPrimary"
        data-block="link to Github"
        href="https://github.com/arshad-shah"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
      <a
        aria-disabled="false"
        aria-label="link to Linkedin account of Arshad shah"
        class="MuiButtonBase-root MuiIconButton-root makeStyles-socialLinks-3 MuiIconButton-colorPrimary"
        data-block="link to Linkedin"
        href="https://www.linkedin.com/in/arshadshah"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
      <a
        aria-disabled="false"
        aria-label="link to mail account of Arshad shah"
        class="MuiButtonBase-root MuiIconButton-root makeStyles-socialLinks-3 MuiIconButton-colorPrimary"
        data-block="link to Mail"
        href="mailto:shaharshad1999@outlook.com"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </div>
    <button
      aria-label="scroll down button"
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
      data-block="scroll down"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <a>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11 5v11.17l-4.88-4.88c-.39-.39-1.03-.39-1.42 0-.39.39-.39 1.02 0 1.41l6.59 6.59c.39.39 1.02.39 1.41 0l6.59-6.59c.39-.39.39-1.02 0-1.41a.9959.9959 0 00-1.41 0L13 16.17V5c0-.55-.45-1-1-1s-1 .45-1 1z"
            />
          </svg>
        </a>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
